hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/compact/DeserializedGenericRecordBuilder.java

```diff
@@ -32,27 +33,34 @@
  */
 public class DeserializedGenericRecordBuilder extends AbstractGenericRecordBuilder {
 
-    private final TreeMap<String, Object> objects = new TreeMap<>();
+    private final TreeMap<String, Object> objects;
     private final SchemaWriter schemaWriter;
+    private boolean built;
 
     public DeserializedGenericRecordBuilder(String typeName) {
+        this.objects = new TreeMap<>();
         schemaWriter = new SchemaWriter(typeName);
     }
 
     /**
      * @return newly created GenericRecord
      * @throws HazelcastSerializationException if a field is not written when building with builder from
-     *                                         {@link GenericRecordBuilder#compact(String)} (ClassDefinition)} and
+     *                                         {@link GenericRecordBuilder#portable(ClassDefinition)} and
      *                                         {@link GenericRecord#newBuilder()}
      */
     @Nonnull
     @Override
     public GenericRecord build() {
-        return new DeserializedGenericRecord(schemaWriter.build(), objects);
+        GenericRecord record = new DeserializedGenericRecord(schemaWriter.build(), objects);
+        this.built = true;
+        return record;
```

Comment:
```suggestion
        this.built = true;
        return new DeserializedGenericRecord(schemaWriter.build(), objects);;
```
minor

Reply:
I think I did this because I thought some method may throw in `DeserializedGenericRecord(schemaWriter.build(), objects);` but seems like I am wrong. Applying..

Reply:
I think it was due to the throws statement in the Javadoc. I think I should get rid of it as well.